### PR TITLE
fix: Fix issue with invalid sql generated by ml distance functions

### PR DESCRIPTION
**Pull request by TrevorBergeron**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: bigframes/ml/sql.py [hunk 1]
```diff
@@ -170,12 +170,11 @@ def ml_distance(
         col_x: str,
         col_y: str,
         type: Literal["EUCLIDEAN", "MANHATTAN", "COSINE"],
-        source_df: bpd.DataFrame,
+        source_sql: str,
```
Comment (GarrettWu) [inline bigframes/ml/sql.py:173]: Is it needed to make this change?

If possible I would rather to let SQL generations stay in sql.py.

Comment (TrevorBergeron) [reply bigframes/ml/sql.py:173]: Generally, want to invert the dependency between domain objects and sql. Will need to further factor things into a common sql generation module anyways (ml and core bigframes have separate sql modules)

File: bigframes/ml/core.py [hunk 2]
```diff
@@ -35,11 +35,27 @@ def __init__(self, session: bigframes.Session):
...
+    """
+
+    def __init__(self, session: bigframes.Session, model: bigquery.Model):
+        self._session = session
+        self._model = model
+        self._model_manipulation_sql_generator = ml_sql.ModelManipulationSqlGenerator(
+            self.model_name
+        )
+
+    def _predict_sql(
```
[the remark this answers is not in the document]
Comment (TrevorBergeron) [reply bigframes/ml/core.py:53]: I'm open to suggestions, previous name was very generic. Seems that the class of methods is tvfs that take a tables and add a new transform/prediction/classifiction result column from values in the input table.

[the remark this answers is not in the document]
Comment (GarrettWu) [reply bigframes/ml/core.py:53]: What it does is actually executing an ML function and return output as a dataframe. Maybe _exec_ml_func? predict is misleading.

**Review by GarrettWu — APPROVED** [review]

**Pull request merged**